src/components/SelectOptionalCoursesModal.tsx

```diff
@@ -0,0 +1,119 @@
+import {
+  Autocomplete,
+  Box,
+  Button,
+  Checkbox,
+  Grid,
+  Modal,
+  TextField,
+  Typography,
+} from '@mui/material'
+import { useState } from 'react'
+import { OptionalCourse } from '@/types'
+import { useFetchOptionalCoursesQuery } from '@/api'
+
+import CheckBoxOutlineBlankIcon from '@mui/icons-material/CheckBoxOutlineBlank'
+import CheckBoxIcon from '@mui/icons-material/CheckBox'
+import { colors } from '@/theme'
+
+const icon = <CheckBoxOutlineBlankIcon fontSize="small" />
+const checkedIcon = <CheckBoxIcon fontSize="small" />
+
+interface Props {
+  curriculaeId: string
+  open: boolean
+  handleClose: () => void
+  selectedOptionalCourses: OptionalCourse[]
+  setSelectedOptionalCourses: (courses: OptionalCourse[]) => void
+}
+
+export default function SelectOptionalCoursesModal({
+  open,
+  handleClose,
+  curriculaeId,
+  selectedOptionalCourses,
+  setSelectedOptionalCourses,
+}: Props) {
+  const [inputValue, setInputValue] = useState('')
+  const { data: optionalCourses = [] } =
+    useFetchOptionalCoursesQuery(curriculaeId)
+
+  return (
+    <Modal open={open} disableEnforceFocus onClose={handleClose}>
+      <Box
+        sx={{
+          position: 'absolute',
+          top: '50%',
+          left: '50%',
+          transform: 'translate(-50%, -50%)',
+          background: 'white',
+          width: '350px',
+          border: '0',
+          borderRadius: '3px',
+          display: 'flex',
+          flexDirection: 'column',
+          justifyContent: 'center',
+          padding: '16px',
+        }}
+      >
```

Comment:
Aqui acho melhor utilizar as props do componente.

https://smartdevpreneur.com/when-to-use-the-mui-sx-prop-styled-api-or-theme-override/